litellm/llms/vertex_httpx.py

```diff
@@ -748,10 +748,12 @@ def load_auth(
             if project_id is None:
                 project_id = creds.project_id
         else:
-            creds, project_id = google_auth.default(
+            creds, creds_project_id = google_auth.default(
                 quota_project_id=project_id,
                 scopes=["https://www.googleapis.com/auth/cloud-platform"],
             )
+            if project_id is None:
+                project_id = creds_project_id
```

Comment:
Hi @krrishdholakia, I saw the change in 

> hey @t968914 should be fixed already with this - [6b14cf7](https://github.com/BerriAI/litellm/commit/6b14cf765708376490c5d88d3e54edc173c343b6)

This part is in the else block and for the application default credential case when a service account isn't being used. The google auth default method will always return None for project_id in this case, and similarly write over the previous value passed into this method.

Reply:
Our models in production are also deployed in a different GCP project than our litellm proxy, so we are unable to use vertex_ai_beta in production either

Reply:
noted - working on this today

Reply:
thanks for the review @andrewmjc @t968914